### PR TITLE
QoL: Updating all the provider Golang code to include `Description` in preparation of documentation generation

### PR DESCRIPTION
**Pull request by detro**
This is extrapolated by #167, but makes the change-set more "manageable" for a good review process.

Additional to the addition of a lot of markdown-based `Description` strings to the `Schema`s, this slightly re-organises the common part of the `*_cert.go` resources, so their schemas are set by the same utility methods (see `common_cert.go`).

I have explicitly used `git mv resource_certificate.go common_cert.go` but somehow git decided it's still not a move and created a delete and a create 😢

### CODE REVIEW

File: internal/provider/common_cert.go [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+	}
+
+	hash := sha1.Sum(publicKeyBytes)
+	return hash[:], nil
+}
+
+// setCertificateSubjectSchema sets on the given reference to map of schema.Schema
+// all the keys required by a resource representing a certificate's subject.
+func setCertificateSubjectSchema(s *map[string]*schema.Schema) {
+	(*s)["dns_names"] = &schema.Schema{
```
Comment (bflad) [inline internal/provider/common_cert.go:94]: Nit: Using map pointers is extraneous -- these can be simplified to 

```go
func setCertificateSubjectSchema(s map[string]*schema.Schema) {
	s["dns_names"] = &schema.Schema{
```

etc. 👍

File: internal/provider/common_cert.go [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+		Description: "The subject for which a certificate is being requested. " +
+			"The acceptable arguments are all optional and their naming is based upon " +
+			"[Issuer Distinguished Names (RFC5280)](https://tools.ietf.org/html/rfc5280#section-4.1.2.4) section.",
+	}
+}
+
+// setCertificateCommonSchema sets on the given reference to map of schema.Schema
+// all the keys required by a resource representing a certificate.
+func setCertificateCommonSchema(s *map[string]*schema.Schema) {
+	(*s)["validity_period_hours"] = &schema.Schema{
```
Comment (bflad) [inline internal/provider/common_cert.go:220]: Similarly

```go
func setCertificateCommonSchema(s map[string]*schema.Schema) {
	s["validity_period_hours"] = &schema.Schema{
```

**Review by bflad — APPROVED** [review]
Some minor considerations, otherwise looks good to me 🚀

File: internal/provider/common_cert.go (hunk 1 above)
Comment (detro) [reply internal/provider/common_cert.go:94]: Yeah, I forgot to integrate that feedback. Will do now.

File: internal/provider/common_cert.go [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+		Description: "Is the generated certificate representing a Certificate Authority (CA) (default: `false`).",
+	}
+
+	(*s)["allowed_uses"] = &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		ForceNew: true,
+		Elem: &schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validation.StringInSlice(supportedKeyUsages(), false),
```
[the remark this answers is not in the document]
Comment (detro) [reply internal/provider/common_cert.go:255]: Yes, I think it's worth starting populating the CHANGELOG for 3.2.0.

**Pull request merged**